frontend/src/store/credential.js

```diff
@@ -0,0 +1,218 @@
+        enumerable: false,
+      })
+
+      const quotas = (secretBinding.quotas || [])
+        .map(quota => get(state.quotas, namespaceNameKey(quota)))
+        .filter(Boolean)
+
+      Object.defineProperty(secretBinding, '_quotas', {
+        value: quotas,
+        configurable: true,
+        enumerable: false,
+      })
+
+      return secretBinding
+    })
+  })
+
+  const secretList = computed(() => {
+    return Object.values(state.secrets)
+  })
+
+  const quotaList = computed(() => {
+    return Object.values(state.quotas)
+  })
+
+  async function createCredential (params) {
+    const { data: { secretBinding, secret } } = await api.createCloudProviderCredential({ secretBinding: params.secretBinding, secret: params.secret })
+    updateCloudProviderCredential({ secretBinding, secret })
+    appStore.setSuccess('Cloud Provider credential created')
+  }
+
+  async function updateCredential (params) {
+    const { data: { secretBinding, secret } } = await api.updateCloudProviderCredential({ secretBinding: params.secretBinding, secret: params.secret })
+    updateCloudProviderCredential({ secretBinding, secret })
+    appStore.setSuccess('Cloud Provider credential updated')
+  }
+
+  async function deleteCredential (name) {
+    const namespace = authzStore.namespace
+
+    await api.deleteCloudProviderCredential({ namespace, name })
+    remove({ namespace, name })
+    appStore.setSuccess('Cloud Provider credential deleted')
+  }
+
+  const infrastructureSecretBindingsList = computed(() => {
+    return filter(secretBindingList.value, secretBinding => {
+      return cloudProfileStore.sortedProviderTypeList.includes(secretBinding.provider?.type)
+    })
+  })
+
+  const dnsSecretBindingsList = computed(() => {
+    return filter(secretBindingList.value, secretBinding => {
+      return gardenerExtensionStore.dnsProviderTypes.includes(secretBinding.provider?.type) && isOwnSecret(secretBinding) // setting secret binding not supported
+    })
+  })
+
+  function getSecret ({ namespace, name }) {
+    return get(state.secrets, namespaceNameKey({ namespace, name }))
+  }
+
+  function getSecretBinding ({ namespace, name }) {
+    return get(state.secretBindings, namespaceNameKey({ namespace, name }))
+  }
+
+  function updateCloudProviderCredential ({ secretBinding, secret }) {
+    const key = namespaceNameKey(secretBinding.metadata)
+    set(state.secretBindings, [key], secretBinding)
+
+    if (secret) {
+      const key = namespaceNameKey(secret.metadata)
+      set(state.secrets, [key], secret)
+    }
+
+    // no update logic for quotas as they currently cannot be updated using the dashboard
+  }
+
+  function remove ({ namespace, name }) {
+    const secretBinding = getSecretBinding({ namespace, name })
+    const secretRef = secretBinding.secretRef
+    const secretBindingKey = namespaceNameKey(secretBinding.metadata)
+
+    delete state.secretBindings[secretBindingKey] // eslint-disable-line security/detect-object-injection
```

Comment:
always include a short comment when disabling eslint security checks (also for the next occurrences in this file)